ducktape/tests/session.py

```diff
@@ -39,6 +39,7 @@ def __init__(self, session_id, results_dir, cluster, args):
         self.args = args
         self.debug = args.debug
         self.exit_first = args.exit_first
+        self.no_teardown = args.no_teardown
```

Comment:
I'm just noticing that we're relying on argparse-style objects being passed in here. That's a bit unusual since this style wouldn't work for normal dict/kwargs-style argument passing for sessions.

Maybe just file a followup issue that should turn these into kwargs or something? I think it might not be hard to have the parsed args passed in that way.